src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -810,6 +813,53 @@ private SqlCommand BuildGetChangesCommand(SqlConnection connection, SqlTransacti
             return new SqlCommand(getChangesQuery, connection, transaction);
         }
 
+        /// <summary>
+        /// Returns the query to get for number of changes(rows) on the user's table that are actively locked by other leases.
+        /// </summary>
+        /// <param name="connection">The connection to add to the returned SqlCommand</param>
+        /// <param name="transaction">The transaction to add to the returned SqlCommand</param>
+        /// <returns>The number of rows locked by leases</returns>
+        private async Task<int> GetLeaseLockedRowCount(SqlConnection connection, SqlTransaction transaction)
+        {
+            string userTableJoinCondition = string.Join(" AND ", this._primaryKeyColumns.Select(col => $"c.{col.name.AsBracketQuotedString()} = u.{col.name.AsBracketQuotedString()}"));
```

Comment:
We shouldn't need to join on the user table, right? Just getting the number of changes that are locked should be enough.